src/mam4xx/hetfrz.hpp

```diff
@@ -1234,25 +1234,25 @@ void hetfrz_rates_1box(const int k, const Real dt, const Atmosphere &atm,
   const int accum_idx = int(ModeIndex::Accumulation);
   const int pcarbon_idx = int(ModeIndex::PrimaryCarbon);
 
-  auto &accum_bc = progs.q_aero_i[accum_idx][int(AeroId::BC)];
-  auto &accum_dst = progs.q_aero_i[accum_idx][int(AeroId::DST)];
-  auto &accum_soa = progs.q_aero_i[accum_idx][int(AeroId::SOA)];
-  auto &accum_so4 = progs.q_aero_i[accum_idx][int(AeroId::SO4)];
-  auto &accum_pom = progs.q_aero_i[accum_idx][int(AeroId::POM)];
-  auto &accum_mom = progs.q_aero_i[accum_idx][int(AeroId::MOM)];
+  auto &accum_bc  = progs.q_aero_i[accum_idx][aerosol_index_for_mode(ModeIndex::Accumulation, AeroId::BC)];
```

Comment:
This change is needed. However, I do not like the fact that we need to use ` ModeIndex::Accumulation` twice in progs. Should we create an issue to improve progs?

Reply:
I agree, we can look into it to see how to fix this. I will create an issue.